akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -87,69 +88,82 @@ object AkkaHttpAdapter extends FailFastCirceSupport {
+              case "connection_init" =>
+                IO.fromFuture(_ => queue.offer(TextMessage("""{"type":"connection_ack"}""")))
+              case "connection_terminate" =>
+                IO.effect(queue.complete())
+              case "start" =>
+                val payload = msg.hcursor.downField("payload")
+                val id      = msg.hcursor.downField("id").success.flatMap(_.value.asString).getOrElse("")
+                Task.whenCase(payload.downField("query").success.flatMap(_.value.asString)) {
+                  case Some(query) =>
+                    val operationName = payload.downField("operationName").success.flatMap(_.value.asString)
+                    (for {
+                      result <- execute(interpreter, GraphQLRequest(query, operationName, None))
+                      _ <- result.data match {
+                            case ObjectValue((fieldName, StreamValue(stream)) :: Nil) =>
+                              stream.foreach { item =>
+                                sendMessage(queue, id, ObjectValue(List(fieldName -> item)), result.errors)
+                              }.fork.flatMap(fiber => subscriptions.update(_.updated(id, fiber)))
+                            case other =>
+                              sendMessage(queue, id, other, result.errors) *> IO.fromFuture(
+                                _ => queue.offer(TextMessage(s"""{"type":"complete","id":"$id"}"""))
+                              )
+                          }
+                    } yield ()).catchAll(
+                      error =>
+                        IO.fromFuture(
+                          _ =>
+                            queue.offer(
+                              TextMessage(
+                                Json
+                                  .obj(
+                                    "id"      -> Json.fromString(id),
+                                    "type"    -> Json.fromString("complete"),
+                                    "payload" -> Json.fromString(error.toString)
+                                  )
+                                  .noSpaces
+                              )
+                          )
+                      )
+                    )
+                }
+              case "stop" =>
+                val id = msg.hcursor.downField("id").success.flatMap(_.value.asString).getOrElse("")
+                subscriptions
+                  .modify(map => (map.get(id), map - id))
+                  .flatMap(fiber => IO.whenCase(fiber) { case Some(fiber) => fiber.interrupt })
+            }
+      } yield ()
+
     import akka.http.scaladsl.server.Directives._
 
     get {
       extractUpgradeToWebSocket { upgrade =>
         val (queue, source) = Source.queue[Message](0, OverflowStrategy.fail).preMaterialize()
-        val subscriptions   = runtime.unsafeRun(Ref.make(Map.empty[String, Fiber[Throwable, Unit]]))
+        val subscriptions: Ref[Map[String, Fiber[Throwable, Unit]]] =
```

Comment:
I think there's no need to explicitly put the type here

Reply:
yes, fixed.